source/extensions/filters/common/expr/custom_cel/example/custom_cel_functions.cc

```diff
@@ -0,0 +1,43 @@
+#include "source/extensions/filters/common/expr/custom_cel/example/custom_cel_functions.h"
+
+#include "eval/public/cel_value.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Expr {
+namespace Custom_Cel {
+namespace Example {
+
+absl::Status getProductCelFunction::Evaluate(absl::Span<const CelValue> args, CelValue* output,
+                                             Protobuf::Arena*) const {
+  int64_t value = args[0].Int64OrDie() * args[1].Int64OrDie();
```

Comment:
Yeah, echoing the above point, it would be good to demonstrate utility here though something less artificial. Maybe something like https://github.com/envoyproxy/envoy/issues/19724? Although arguably that should be in core. @kyessenov any thoughts?

Reply:
I agree showing the utility by implementing request.query, unfortunately one (reasonable) limitation of CEL is that you can't mix-and-match attributes implemented separately. the 'RequestWrapper' object that implements request.path etc must be the one that implements all request.* attributes.  We could support request.query() or query(request) here though.

Related, the toy functions etc here I believe are used to display support for attributes, functions, and lazy functions, are you looking for utility in each of these examples, or just one?

Reply:
I was going to copy and paste the existing RequestWrapper and 'query' to it.  

Weston suggested extending RequestWrapper.

Reply:
I think it will be fine to just have a single extension that does something that an Envoy user would potentially use. The problem with pure example extensions is it's unclear why we even have the extension point.